public/app/features/gops/configuration-tracker/alerting/hooks.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect } from 'react';
+
+import { alertRuleApi } from 'app/features/alerting/unified/api/alertRuleApi';
+import { alertmanagerApi } from 'app/features/alerting/unified/api/alertmanagerApi';
+import { useDataSourceFeatures } from 'app/features/alerting/unified/hooks/useCombinedRule';
+import { GRAFANA_RULES_SOURCE_NAME } from 'app/features/alerting/unified/utils/datasource';
+import { Receiver } from 'app/plugins/datasource/alertmanager/types';
+
+export function useIsCreateAlertRuleDone() {
+  const [fetchRulerRules, { data, isLoading }] = alertRuleApi.endpoints.rulerRules.useLazyQuery({
```

Comment:
Perhaps it would be more efficient to simply check if we have at least one rule from the prometheus endpoint for Grafana-managed, we can limit the response size with

`/api/prometheus/grafana/api/v1/rules?limit_alerts=1&limit=1&limit_rules=1`

Which is significantly smaller in size :D

Reply:
good point. I started using promehteus endpoint (without the LIMIT), and @konrad147 suggested me to switch to ruler api as its faster than prometheus.

you can see the change in this [commit](https://github.com/grafana/grafana/pull/85838/commits/5c68c5bb8a05ea0e96edc65b31bf8a3284cb4a58)